test/e2e-go/features/accountPerf/fourMillion_test.go

```diff
@@ -0,0 +1,355 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package fataccount
+
+import (
+	"fmt"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/libgoal"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+const roundDelay = uint64(200)
+
+func queuePayments(queueWg *sync.WaitGroup, c libgoal.Client, q <-chan *transactions.SignedTxn) {
+	counter := 0
+	for stxn := range q {
+
+		if stxn == nil {
+			break
+		}
+		for {
+			_, err := c.BroadcastTransaction(*stxn)
+			if err == nil {
+				counter++
+				break
+			}
+			fmt.Printf("Error broadcasting transaction: %v\n", err)
+			time.Sleep(config.Consensus[protocol.ConsensusCurrentVersion].AgreementFilterTimeout)
+		}
+	}
+	queueWg.Done()
+}
+
+func signer(t *testing.T, sigWg *sync.WaitGroup, client libgoal.Client, txnChan <-chan *transactions.Transaction, sigTxnChan chan<- *transactions.SignedTxn) {
+	for txn := range txnChan {
+		if txn == nil {
+			break
+		}
+		walletHandle, err := client.GetUnencryptedWalletHandle()
+		require.NoError(t, err)
+
+		stxn, err := client.SignTransactionWithWallet(walletHandle, nil, *txn)
+		if err != nil {
+			fmt.Printf("Error signing: %v\n", err)
+		}
```

Comment:
There are two issues here that would kill the performance:
1. Since you're using a devmode network, please try to use transaction groups of size 16. It would make each block contain these 16 transactions.
2. avoid using the KMD. It won't be sufficiently performant.
3. when sending transaction, use the BroadcastTransactionGroup.

Reply:
Done!